sdk/core/Azure.Core/tests/MockHttpClientHandler.cs

```diff
@@ -32,5 +32,14 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
 
             return response ?? new HttpResponseMessage((HttpStatusCode)200);
         }
+
+#if NET5_0
+        protected override HttpResponseMessage Send(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = _onSend(request).GetAwaiter().GetResult();
```

Comment:
`EnsureCompleted`?

Reply:
This is a test implementation that explicitly blocks because the task is not guaranteed to be completed.